### PR TITLE
Move ModuleFileSystem and ModuleHardware to utils

### PR DESCRIPTION
**Pull request by alvarosoppelsa**
- Remove modules from Application class
- Rename as Hardware and FileSystem
- Move to utils/
- Fix dependencies

### CODE REVIEW

**Review by miquelmiro3 — APPROVED** [review]

File: Source/src/ui/WindowAbout.cpp [hunk 1]
```diff
@@ -16,23 +22,23 @@ void Hachiko::WindowAbout::Update()
         ImGui::Text("Engine name: %s", TITLE);
         ImGui::Text("Version: %s", ENGINE_VERSION);
         ImGui::Separator();
-        ImGui::Text("Eric Torres Perramon & Bernat UWU");
+        ImGui::Text("Akita Interactive Studios");
```
Comment (baransrc) [inline Source/src/ui/WindowAbout.cpp:25]:
```suggestion
        ImGui::Text("Akita Interactive");
```

File: Source/src/ui/WindowAbout.cpp [hunk 2]
```diff
@@ -16,23 +22,23 @@ void Hachiko::WindowAbout::Update()
...
-        ImGui::Text("GPU: %s", App->hw->GetInfo().gpu);
-        ImGui::Text("Brand: %s", App->hw->GetInfo().gpu_brand);
-        ImGui::Text("VRAM Budget: %.1f Mb", static_cast<float>(App->hw->GetInfo().vram_mb_budget));
-        ImGui::Text("Vram Avaliable:  %.1f Mb", static_cast<float>(App->hw->GetInfo().vram_mb_free));
-        ImGui::Text("Vram Usage:  %.1f Mb", static_cast<float>(App->hw->GetInfo().vram_mb_budget - App->hw->GetInfo().vram_mb_free));
+        ImGui::Text("GPU: %s", hw->GetInfo().gpu);
+        ImGui::Text("Brand: %s", hw->GetInfo().gpu_brand);
+        ImGui::Text("VRAM Budget: %.1f Mb", static_cast<float>(hw->GetInfo().vram_mb_budget));
+        ImGui::Text("Vram Avaliable:  %.1f Mb", static_cast<float>(hw->GetInfo().vram_mb_free));
+        ImGui::Text("Vram Usage:  %.1f Mb", (float)(hw->GetInfo().vram_mb_budget) - (float)(hw->GetInfo().vram_mb_free));
```
Comment (baransrc) [inline Source/src/ui/WindowAbout.cpp:41]: Can we rename `hw` to `hardware`?

File: Source/src/ui/WindowAbout.h [hunk 3]
```diff
@@ -4,12 +4,17 @@
...
     class WindowAbout final : public Window
     {
     public:
         WindowAbout();
 
         ~WindowAbout() override;
         void Update() override;
+
+    private:
+        Hardware* hw;
```
Comment (baransrc) [inline Source/src/ui/WindowAbout.h:18]:
```suggestion
        Hardware* hardware;
```

File: Source/src/utils/FileSystem.cpp [hunk 4]
```diff
@@ -14,24 +15,23 @@ Hachiko::ModuleFileSystem::ModuleFileSystem()
...
 {
     HE_LOG("Init virtual file system");
 
     CreateContext();
 
     char* write_path = SDL_GetPrefPath("Akita", "Hachiko");
     SDL_free(write_path);
-
-    return true;
+    return false;
```
Comment (Erick9Thor) [inline Source/src/utils/FileSystem.cpp:31]: This return value is correct?

**Review by Erick9Thor — APPROVED** [review]

**Review by baransrc — COMMENTED** [review]
Thanks for the improvements! I've requested a very small change, after that, we are good to go.

File: Source/src/ui/WindowAbout.cpp (hunk 1 above)
Comment (alvarosoppelsa) [reply Source/src/ui/WindowAbout.cpp:25]: Done!

File: Source/src/ui/WindowAbout.cpp (hunk 2 above)
Comment (alvarosoppelsa) [reply Source/src/ui/WindowAbout.cpp:41]: Sure, better

File: Source/src/ui/WindowAbout.cpp (hunk 1 above)
Comment (baransrc) [reply Source/src/ui/WindowAbout.cpp]: ❤️

**Review by baransrc — APPROVED** [review]
Everything LGTM!

File: Source/src/utils/FileSystem.cpp (hunk 4 above)
Comment (alvarosoppelsa) [reply Source/src/utils/FileSystem.cpp:31]: Nope. But as is not a Module anymore, Application does not validate its return value. I can modifying it to return true, or change the function to return void. What is better for you?

Comment (alvarosoppelsa) [reply Source/src/utils/FileSystem.cpp:31]: Well seeing btw

Comment (Erick9Thor) [reply Source/src/utils/FileSystem.cpp:31]: it's fine for me if this function never is asserted

Comment (alvarosoppelsa) [reply Source/src/utils/FileSystem.cpp:31]: Ok!

**Pull request merged**